service/create/cloud_config.go

```diff
@@ -93,17 +76,264 @@ func (c *CloudConfigExtension) Units() ([]cloudconfig.UnitAsset, error) {
+			Encoding:     cloudconfig.GzipBase64,
+			Permissions:  0700,
+		},
+		cloudconfig.FileMetadata{
+			AssetContent: m.TLSAssets.APIServerCA,
+			Path:         "/etc/kubernetes/ssl/apiserver-ca.pem.enc",
+			Owner:        "root:root",
+			Encoding:     cloudconfig.GzipBase64,
+			Permissions:  0700,
+		},
+		cloudconfig.FileMetadata{
+			AssetContent: m.TLSAssets.APIServerKey,
+			Path:         "/etc/kubernetes/ssl/apiserver-key.pem.enc",
+			Owner:        "root:root",
+			Encoding:     cloudconfig.GzipBase64,
+			Permissions:  0700,
+		},
+		cloudconfig.FileMetadata{
+			AssetContent: m.TLSAssets.ServiceAccountCrt,
+			Path:         "/etc/kubernetes/ssl/service-account-crt.pem.enc",
+			Owner:        "root:root",
+			Encoding:     cloudconfig.GzipBase64,
+			Permissions:  0700,
+		},
+		cloudconfig.FileMetadata{
+			AssetContent: m.TLSAssets.ServiceAccountCA,
+			Path:         "/etc/kubernetes/ssl/service-account-ca.pem.enc",
+			Owner:        "root:root",
+			Encoding:     cloudconfig.GzipBase64,
+			Permissions:  0700,
+		},
+		cloudconfig.FileMetadata{
+			AssetContent: m.TLSAssets.ServiceAccountKey,
+			Path:         "/etc/kubernetes/ssl/service-account-key.pem.enc",
+			Owner:        "root:root",
+			Encoding:     cloudconfig.GzipBase64,
+			Permissions:  0700,
+		},
+		cloudconfig.FileMetadata{
+			AssetContent: m.TLSAssets.CalicoClientCrt,
+			Path:         "/etc/kubernetes/ssl/calico/client-crt.pem.enc",
+			Owner:        "root:root",
+			Encoding:     cloudconfig.GzipBase64,
+			Permissions:  0700,
+		},
+		cloudconfig.FileMetadata{
+			AssetContent: m.TLSAssets.CalicoClientCA,
+			Path:         "/etc/kubernetes/ssl/calico/client-ca.pem.enc",
+			Owner:        "root:root",
+			Encoding:     cloudconfig.GzipBase64,
+			Permissions:  0700,
+		},
+		cloudconfig.FileMetadata{
+			AssetContent: m.TLSAssets.CalicoClientKey,
+			Path:         "/etc/kubernetes/ssl/calico/client-key.pem.enc",
+			Owner:        "root:root",
+			Encoding:     cloudconfig.GzipBase64,
+			Permissions:  0700,
+		},
+		cloudconfig.FileMetadata{
+			AssetContent: m.TLSAssets.EtcdServerCrt,
+			Path:         "/etc/kubernetes/ssl/etcd/server-crt.pem.enc",
+			Owner:        "root:root",
+			Encoding:     cloudconfig.GzipBase64,
+			Permissions:  0700,
+		},
+		cloudconfig.FileMetadata{
+			AssetContent: m.TLSAssets.EtcdServerCA,
+			Path:         "/etc/kubernetes/ssl/etcd/server-ca.pem.enc",
+			Owner:        "root:root",
+			Encoding:     cloudconfig.GzipBase64,
+			Permissions:  0700,
+		},
+		cloudconfig.FileMetadata{
+			AssetContent: m.TLSAssets.EtcdServerKey,
+			Path:         "/etc/kubernetes/ssl/etcd/server-key.pem.enc",
+			Owner:        "root:root",
+			Encoding:     cloudconfig.GzipBase64,
+			Permissions:  0700,
+		},
+	}
+
+	files, err := m.renderFiles(masterFilesMeta)
+	if err != nil {
+		return nil, microerror.MaskAny(err)
+	}
+
+	return files, nil
+}
+
+func (m *MasterCloudConfigExtension) Units() ([]cloudconfig.UnitAsset, error) {
```

Comment:
Couldn't the two `Units` methods be defined on the `CloudConfigExtension` type, since they don't change?